pkg/controller/plan/migration.go

```diff
@@ -1143,129 +1140,122 @@ func (r *Migration) setRunning(vm *plan.VMStatus, running bool) (err error) {
+	if err != nil {
+		return
+	}
+	if dvs == nil || len(dvs) == 0 {
+		pvcs, err = r.kubevirt.getPVCs(vm)
+		if err != nil {
+			return
+		}
+		for _, pvc := range pvcs {
+			var task *plan.Task
+			name := r.builder.ResolvePersistentVolumeClaimIdentifier(&pvc)
+			found := false
+			task, found = step.FindTask(name)
+			if !found {
 				continue
 			}
-
+			if pvc.Status.Phase == core.ClaimBound {
+				completed++
+				task.Phase = Completed
+				task.Reason = "Transfer completed."
+				task.Progress.Completed = task.Progress.Total
+				task.MarkCompleted()
+			}
+		}
+	} else {
+		for _, dv := range dvs {
+			var task *plan.Task
+			name := r.builder.ResolveDataVolumeIdentifier(dv.DataVolume)
+			found := false
+			task, found = step.FindTask(name)
 			if !found {
-				log.Info(
-					"Did not find CDI importer pod for DataVolume.",
-					"vm",
-					vm.String(),
-					"dv",
-					path.Join(dv.Namespace, dv.Name))
 				continue
 			}
 
-			if r.Plan.Spec.Warm && len(importer.Status.ContainerStatuses) > 0 {
-				vm.Warm.Failures = int(importer.Status.ContainerStatuses[0].RestartCount)
-			}
-			if restartLimitExceeded(importer) {
-				task.MarkedCompleted()
-				msg, _ := terminationMessage(importer)
-				task.AddError(msg)
+			conditions := dv.Conditions()
+			switch dv.Status.Phase {
+			case cdi.Succeeded, cdi.Paused:
+				completed++
+				task.Phase = Completed
+				task.Reason = "Transfer completed."
+				task.Progress.Completed = task.Progress.Total
+				task.MarkCompleted()
+			case cdi.Pending, cdi.ImportScheduled:
+				pending++
+				task.Phase = Pending
+				cnd := conditions.FindCondition("Bound")
+				if cnd != nil && cnd.Status == True {
+					cnd = conditions.FindCondition("Running")
+				}
+				if cnd != nil {
+					pendingReason = fmt.Sprintf("%s; %s", cnd.Reason, cnd.Message)
+				}
+				task.Reason = pendingReason
+			case cdi.ImportInProgress:
+				running++
+				task.Phase = Running
+				task.MarkStarted()
+				cnd := conditions.FindCondition("Running")
+				if cnd != nil {
+					task.Reason = fmt.Sprintf("%s; %s", cnd.Reason, cnd.Message)
+				}
+				pct := dv.PercentComplete()
+				completed := pct * float64(task.Progress.Total)
+				task.Progress.Completed = int64(completed)
+
+				// The importer pod is recreated by CDI if it is removed for some
+				// reason while the import is in progress, so we can assume that if
+				// we can't find it or retrieve it for some reason that this will
+				// be a transient issue, and we should be able to find it on subsequent
+				// reconciles.
+				var importer *core.Pod
+				var found bool
+				var kErr error
+				if dv.PVC == nil {
+					found = false
+				} else {
+					importer, found, kErr = r.kubevirt.GetImporterPod(*dv.PVC)
+				}
+				if kErr != nil {
+					log.Error(
+						kErr,
+						"Could not get CDI importer pod for DataVolume.",
+						"vm",
+						vm.String(),
+						"dv",
+						path.Join(dv.Namespace, dv.Name))
+					continue
+				}
+
+				if !found {
```

Comment:
hitting found == false many times because the dv.PVC is nil. maybe worth querying the PVC here? it just spam in the logs, the data transfer progress does updates.